program/src/stake_account.rs

```diff
@@ -198,17 +198,19 @@ impl StakeAccount {
         // to true. See also https://github.com/ChorusOne/solido/issues/184#issuecomment-861653316.
         let fix_stake_deactivate = true;
 
-        let (active_lamports, activating_lamports, deactivating_lamports) = stake
+        let (mut active_lamports, activating_lamports, deactivating_lamports) = stake
             .delegation
             .stake_activating_and_deactivating(target_epoch, history, fix_stake_deactivate);
 
-        // Deactivating will be counted in the active lamports
-        let inactive_lamports = account_lamports
-            .0
+        // Deactivating lamports is counted on the active lamports.
+        active_lamports -= deactivating_lamports;
```

Comment:
Aaah, now I understand what you meant ... so Solana’s `stake_activating_deactivating` counts the same lamports twice, once as part of `active_lamports`, and once as part of `deactivating_lamports`?

That’s unexpected to me, but then again, I don’t know what I was expecting, the function is not documented at all :weary: 

Great idea to add a test to confirm the behavior!